### PR TITLE
fix integration with model repository and update README

### PR DESCRIPTION
**Pull request by BertKleewein**

### CODE REVIEW

File: sdk/digitaltwins/dtdl-parser/src/parser/modelParserImpl.ts [hunk 1]
```diff
@@ -205,12 +205,10 @@ export class ModelParserImpl implements ModelParser {
...
 
-      let dependencyMapping: { [dtmi: string]: unknown } | null;
-      try {
-        dependencyMapping = await this.getModels(undefinedIdentifiers);
-      } catch (e) {
-        dependencyMapping = null;
-      }
+      const dependencyMapping = await this.getModels(undefinedIdentifiers, {
+        dependencyResolution: "enabled"
+      });
```
Comment (BertKleewein) [inline sdk/digitaltwins/dtdl-parser/src/parser/modelParserImpl.ts:210]: 2 questions:
1. this catch hid a mistake that I made (getModels not a function). I think we're better without the catch, right?
2. Is this the right dependency resolution mode?

File: sdk/digitaltwins/dtdl-parser/README.md [hunk 2]
```diff
@@ -53,24 +53,36 @@ To parse a sample DTDL model, either start with one you have already written or
...
-  Object.entries(models).forEach(([key, value]) => {
-    console.log(`dtmi: ${key}`);
-    const modelDict = await modelParser.parse([value]);
-    Object.entries(modelDict).forEach(([key2, value2]) => {
-      console.log(key2);
-    });
-  });
+
+  // Bind the model repository to the parser so the parser can use it to load
+  // any dependant models.
```
Comment (BertKleewein) [inline sdk/digitaltwins/dtdl-parser/README.md:66]: >/ any dependant models. [](http://example.com/codeflow?start=3&length=23)

Please review the comments in this example for correct use of terms (specifically "dependant models" and "the contents of each model")

File: sdk/digitaltwins/dtdl-parser/src/parser/modelParserImpl.ts (hunk 1 above)
Comment (YoDaMa) [reply sdk/digitaltwins/dtdl-parser/src/parser/modelParserImpl.ts:210]: can't remember why I implemented it like that with the catch, but yeah as long as it isn't an unhandled promise rejection on failure.

**Review by YoDaMa — APPROVED** [review]
lgtm

**Pull request merged**